pulsar-broker/src/main/java/org/apache/pulsar/broker/service/SystemTopicBasedTopicPoliciesService.java

```diff
@@ -373,6 +361,18 @@ private void initPolicesCache(SystemTopicClient.Reader<PulsarEvent> reader, Comp
         });
     }
 
+    private void cleanCacheAndCloseReader(NamespaceName namespace, boolean cleanOwnedBundlesCount) {
```

Comment:
Do we need to consider the thread-safe issue here?

Reply:
Move `policyCacheInitMap.remove(namespace)` to the end of this method to ensure the state.

Reply:
@codelipenghui 

Threading issues will be caused by the point in time when reader failures and new reader creations occur at the same time.
We create a new Reader using the "prepareInitPoliciesCache" method, which uses the "ConcurrentHashMap" to ensure that the "policyCacheInitMap" is atomic.
Therefore, we can clean this atomic value last when cleaning to ensure that cleanup and new creation are atomic.
I'm not sure if this is a thread safety issue as you think. Please let me know if I understand wrong.